File: Seccion 11 - Funciones en C++/Seccion_11_Python/26_ejercicio_20.py

import typing

def fibonacci(n: int) -> int:
    if n < 2:
        return n
    else:
        return fibonacci(n - 1) + fibonacci(n - 2)

nElementos = 0

while (nElementos <= 0):
    nElementos = int(input('Digite el número de elementos: '))
    
    if nElementos <= 0:
        print('El número de elementos debe ser un entero positivo, no cero!!')

print('\nSerie Fibonacci para n = {}'.format(nElementos))

for i in range(nElementos):
    print('{}, '.format(fibonacci(i)), end='')

print()

File: Seccion 11 - Funciones en C++/Seccion_11_Dart/26_ejercicio_20.dart

import 'dart:io';

int fibonacci(int n) {
  if (n < 2) {
    return n;
  } else {
    return fibonacci(n - 1) + fibonacci(n - 2);
  }
}

void main() {

  int nElementos;

  do {
    stdout.writeln('Digite el número de elementos: ');
    nElementos = int.parse(stdin.readLineSync());

    if (nElementos <= 0) {
      print('El número de elementos debe ser un entero positivo,no cero!!');
    }
  } while (nElementos <= 0);

  print('\nSerie Fibonacci para n = $nElementos');
  for (int i = 0; i < nElementos; i++) {
    stdout.write('${fibonacci(i)}, ');
  }

  print('');
  

}